sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/SaveAsHiveFile.scala

```diff
@@ -140,7 +141,9 @@ private[hive] trait SaveAsHiveFile extends DataWritingCommand {
     try {
       createdTempDir.foreach { path =>
         val fs = path.getFileSystem(hadoopConf)
-        if (fs.delete(path, true)) {
+        // Sometimes (e.g., when speculative task is enabled), temporary directories may be
+        // left uncleaned, confirmTempDirDeleted can confirm deleteOnExit.
+        if (fs.delete(path, true) && !SQLConf.get.confirmTempDirDeleted) {
```

Comment:
If the description is true, we should check if `spark.speculation` is true or false here?

Reply:
@maropu Thanks for reply. In my scenario it was indeed caused by `spark.speculation`, but I don’t know if there are any other scenarios also happened in this.
If it only happened with `spark.speculation`, then we can just check `spark.speculation` instead of adding new configuration.

Reply:
It is happened when running tasks is killed after Job committed, maybe the better way to solve this is checking the job's tasks status. However it seems it is impossible to check status here, thus I add new configuration to this.